core/os/os.cpp

```diff
@@ -306,6 +306,15 @@ String OS::get_system_dir(SystemDir p_dir) const {
 	return ".";
 }
 
+String OS::get_safe_application_name() const {
+	String an = Globals::get_singleton()->get("application/name");
+	Vector<String> invalid_char = String("\\ / : * ? \" < > |").split(" ");
+	for (int i=0;i<invalid_char.size();i++) {
+		an = an.replace(invalid_char[i],"-");
```

Comment:
Yeah, so I guess we should keep what you proposed. At any rate I want to change the code that detected the project user path, to allow more customisable paths, so it will be possible for people to override it if they don't like having ``----`` in their folder names :)